scripts/services/map_manager.gd

```diff
@@ -43,20 +49,26 @@ func get_maps_page(page_number, page_size):
     var map_key_list = self.maps.keys()
     map_key_list.sort()
 
-    if index_end > map_key_list.size():
-        index_end = map_key_list.size()
+    if index_end > _get_maps_count():
+        index_end = _get_maps_count()
 
     var index = index_start
     var output = []
 
     while index < index_end:
-        output.append(map_key_list[index])
+        if index < map_key_list.size():
+            output.append(map_key_list[index])
+        else:
+            output.append(bundled_maps.keys()[index])
```

Comment:
I don't really like the logic here. I think bundled maps should be in front of the list, as these are usually used the most by players. Also doing `map_key_list = bundled_maps.keys().sort() + map_key_list` after `map_key_list` has been sorted would make this loop simpler.